jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/core/assembly/JKubeTarArchiver.java

```diff
@@ -67,7 +67,7 @@ public File createArchive(File inputDirectory, BuildDirs buildDirs, ArchiveCompr
                 String targetFileName = entry.getValue();
 
                 // 1. Check whether nested directory is there, if not create it
-                String[] pathParts = targetFileName.split(File.separator);
+                String[] pathParts = FileUtil.splitFilePathIntoParts(targetFileName);
```

Comment:
Try to use Java native ways to achieve the same purpose.
If I understand correctly, we're trying to create the directory containing the file specified in the `targetFileName` variable.

This can easily be achieved with: `new File(targetFileName).getParent().mkdirs()`, no need to reinvent the wheel or create so many helper class methods (there are also alternatives using the new `java.nio.file`).

_I also see that this `targetFileName` should always point to a file within a directory > don't see a valid compliant use-case having this variable set to something like `/archive.tar` especially on *nix systems > `pathParts.length > 0` shouldn't be optional._